src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Status.all);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('There are no todos'))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const filteredTodos = (todosToFilter: Todo[], filterStatus: Status, searchSymbols: string): Todo[] => {
+    if (filterStatus === Status.all && !searchSymbols.trim()) {
+      return todosToFilter;
+    }
+
+    return todosToFilter.filter(currentTodo => {
+      if (filterStatus === Status.active && currentTodo.completed) {
+        return false;
+      }
+
+      if (filterStatus === Status.completed && !currentTodo.completed) {
+        return false;
+      }
+
+      if (searchSymbols.trim()
+        && !currentTodo.title.toLowerCase()
+          .includes(searchSymbols.toLowerCase())
+      ) {
+        return false;
+      }
```

Comment:
create it as a second filter

firstly create by status

then filter result by query